### PR TITLE
qt6.wrapQtAppsHook: add qtbase and qtwayland to propagatedBuildInputs

### PR DESCRIPTION
**Pull request by wineee**
qtwayland client side is the wayland platform plugin, provides a way to run Qt applications as Wayland clients 

without this, Qt applications only work on x11/xwayland

`wrap-qt5-apps-hook` has make qtwayland propaged
https://github.com/NixOS/nixpkgs/blob/b2063364e120e14ea5a120bd144ab75a1a3b83f5/pkgs/development/libraries/qt-5/5.15/default.nix#L354-L358
## Description of changes

cc @NixOS/qt-kde 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wineee** [comment]
Maybe need more adjusted https://github.com/NixOS/nixpkgs/pull/264964

**Comment by NickCao** [comment]
This was intentionally removed to reduce the closure size of cli only qt programs, but in the long run it looked like a bad idea. Maybe we can have a `wrapQtClisHook` for that particular usecase?

**Comment by wineee** [comment]
1. Not tested on darwin (I don't have the device)
2. Should move libs to depsTargetTargetPropagated(but the [qt5 patch](https://github.com/NixOS/nixpkgs/pull/220518) is still marked as draft, @Artturin Are there still obstacles to doing this?) 
3. qt5.wrapQtAppsHook always propagate qtwayland in linux，maybe we should add `qt5.wrapQtClisHook`

**Comment by Aleksanaa** [comment]
>     3. qt5.wrapQtAppsHook always propagate qtwayland in linux，maybe we should add `qt5.wrapQtClisHook`

This should not matter, because it just happens that there is no qtwayland in qtbase, but logically there should be.

**Comment by K900** [comment]
I think we can leave Qt5 as is at this point, it'll just go away after a while. I'll look at the rest of the change in a bit.

File: pkgs/development/libraries/qt-6/default.nix [hunk 1]
```diff
@@ -182,9 +182,18 @@ let
       };
 
       wrapQtAppsHook = callPackage
-        ({ makeBinaryWrapper }: makeSetupHook
+        ({ makeBinaryWrapper, qtbase, qtwayland }: makeSetupHook
           {
             name = "wrap-qt6-apps-hook";
+            propagatedBuildInputs = [ qtbase.dev makeBinaryWrapper ]
+              ++ lib.optional (lib.meta.availableOn stdenv.targetPlatform qtwayland) qtwayland.dev;
```
Comment (wegank) [inline pkgs/development/libraries/qt-6/default.nix:189]:
```suggestion
              ++ lib.optional (lib.meta.availableOn stdenv.hostPlatform qtwayland) qtwayland.dev;
```

**Comment by drupol** [comment]
It looks like you accidentally mass-pinged a bunch of people, which are now subscribed
and getting notifications for everything in this pull request. Unfortunately, they
cannot be automatically unsubscribed from the issue (removing review request does not
unsubscribe), therefore development cannot continue in this pull request anymore.

Please create a new pull request, link back to this one and ping the
people actually involved in here over there. For the next time, remember to set your PR to draft status before rebasing. In draft status, you can preview the list of maintainers that are about to be requested for review, which allows you to sidestep this issue.

In order to avoid this in the future, there are instructions for how to properly
rebase between branches in our [contribution guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging).
Setting your pull request to draft prior to rebasing is strongly recommended.
In draft status, you can preview the list of people that are about to be requested
for review, which allows you to sidestep this issue.
This is not a bulletproof method, though, as OfBorg still does review requests even on draft PRs.

**Pull request closed without merging**